packages/engine/lib/orchestrator/src/experiment.rs

```diff
@@ -95,6 +95,25 @@ pub struct ExperimentConfig {
     clap(global = true, short = 'w', long, default_value_t = num_cpus::get(), validator = at_least_one, env = "HASH_WORKERS")
     )]
     pub num_workers: usize,
+
+    #[cfg_attr(feature = "clap", clap(global = true, long, default_value_t = 0))]
+    /// Initial size of the V8 heap.
+    ///
+    /// Setting this value allows to avoid garbage collection while the heap is small enough.
```

Comment:
I don't really understand this sentence. If the initial size is set, does that mean, the GC won't ever resize the heap smaller than the provided bytes? Or does it only initialize the size to the specified value?

Reply:
The GC won't run as long as the heap is not bigger than the initial size.
The description above this one is not correct, it was still a bit fuzzy when I wrote it.

Reply:
I changed the other sentence in https://github.com/hashintel/hash/pull/446/commits/5967afd1d09d3af82dcc2a07620d79e23de506ae

Reply:
From the official documentation:

> The initial heap size or zero.
>
> By default V8 starts with a small heap and dynamically grows it to match the set of live objects. This may lead to ineffective garbage collections at startup if the live set is large. Setting the initial heap size avoids such garbage collections. Note that this does not affect young generation garbage collections.

Can't we borrow from that? 

Something like

```
/// The initial heap size for each JavaScript runner in Megabytes.
///
/// By default, each JavaScript runner starts with a small heap and dynamically grows it. The initial heap size can be increased by setting this.
```

If the user is expecting to have a large-scale simulation, it makes sense to increase this value, so v8 doesn't need to reallocate from the heap many times initially.

Reply:
What about
```
/// Allows a JavaScript runner to grow to some initial size in megabytes before triggering garbage collections.
///
/// This is useful when it is known that experiments need a certain minimum heap to run to avoid repeatedly invoking the garbage collector when growing the heap.
```
And maybe a warning that this may waste memory.

Reply:
Addressed in https://github.com/hashintel/hash/pull/446/commits/4995421adf9d244153b456545155ac245df3796b

Reply:
Thank you, I think this is a lot more intuitive!